src/hotspot/share/compiler/compilerOracle.cpp

```diff
@@ -410,15 +410,19 @@ bool CompilerOracle::should_break_at(const methodHandle& method) {
 }
 
 bool CompilerOracle::should_blackhole(const methodHandle& method) {
-  if (check_predicate(CompileCommand::Blackhole, method)) {
-    if (method->result_type() == T_VOID) {
-      return true;
-    } else {
-      warning("blackhole compile command only works for methods with void type: %s",
-              method->name_and_sig_as_C_string());
-    }
+  if (!check_predicate(CompileCommand::Blackhole, method)) {
+    return false;
```

Comment:
Review self-comment. In fact, I am not sure if this should be `check_predicate` (like other commands), or `has_option` (like other options). CompilerOracle refactoring in JDK-8256508 seem to have converged both to mean the same. I left `check_predicate` as in original, and it also matches other `should_*`, see the definitions just above.

Reply:
I think using `check_predicate` is perfectly fine in this case.

Reply:
Okay, I shall leave it at that then.